src/tiled/automapper.cpp

```diff
@@ -631,8 +642,26 @@ static bool layerMatchesConditions(const TileLayer &setLayer,
                         !setLayer.contains(x + offset.x(), y + offset.y()))
                     return false;
 
-                const Cell &setCell = setLayer.cellAt(x + offset.x(),
-                                                      y + offset.y());
+                int xd = x + offset.x();
+                int yd = y + offset.y();
+
+                if (!isMapInfinite) {
+                    if (options.wrapBorder) {
+                        int width = setLayer.size().width();
+                        int height = setLayer.size().height();
+                        xd = (xd % width + width) % width;
+                        yd = (yd % height + height) % height;
+                    } else if (options.overflowBorder) {
+                        if (xd < 0) xd = 0;
+                        else if (xd >= setLayer.size().width())
+                            xd = setLayer.size().width() - 1;
+                        if (yd < 0) yd = 0;
+                        else if (yd >= setLayer.size().height())
+                            yd = setLayer.size().height() - 1;
```

Comment:
Please shorten this to:
```c++
xd = qBound(0, xd, setLayer.width() - 1);
yd = qBound(0, yd, setLayer.height() - 1);
```

Reply:
Thanks! I have never written a line of Qt code before, so I didn't know of that function. That could be very useful, indeed!